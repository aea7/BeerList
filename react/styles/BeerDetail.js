import { StyleSheet } from 'react-native';
import colors from "./Colors";

export default StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: colors.container_background,
        alignItems: 'center',
        marginTop: 25
    },
    text_container: {
        justifyContent: 'space-around',
        paddingTop: 25
    },
    image_style: {
        width: 250,
        height: 250,
        borderRadius: 10,
        marginHorizontal: 10
    },
    field: {
        fontSize: 19,
        fontWeight: 'bold',
        padding: 10,
        color: colors.darker_gray
    },
});