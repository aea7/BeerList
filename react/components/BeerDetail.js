import React from 'react';
import {View, ScrollView, Text, Image} from 'react-native';

import styles from '../styles/BeerDetail'

const Beer = ({navigation}) => {
    let beerItem = navigation.getParam('beerItem');
    return (
        <ScrollView contentContainerStyle={{flexGrow : 1, justifyContent : 'center'}}>
            <View style={styles.container}>
            <Image
                style={styles.image_style}
                source={{uri: beerItem.image_url}}
            />
            <View style={styles.text_container}>
                <Text style={styles.field}> Name: {beerItem.name}</Text>
                <Text style={styles.field}> Description: {beerItem.description}</Text>
                <Text style={styles.field}> Alcohol: {beerItem.abv}%</Text>
                <Text style={styles.field}> Food Pairings: {beerItem.food_pairing}</Text>
            </View>
            </View>
        </ScrollView>
    );

};

export default Beer