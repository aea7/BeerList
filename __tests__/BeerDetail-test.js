import React from 'react';
import BeerDetail from '../react/components/BeerDetail';

import renderer from 'react-test-renderer';
import mockData from "../react/data/mockData";

const beerItem = mockData[0];

test('BeerDetail renders correctly', () => {
    const navigation = { navigate: jest.fn(), getParam: () => beerItem, params: {beerItem: beerItem} };
    navigation.navigate('BeerDetail', {beerItem: beerItem});
    const tree = renderer.create(BeerDetail({navigation})).toJSON();
    expect(tree).toMatchSnapshot();
});