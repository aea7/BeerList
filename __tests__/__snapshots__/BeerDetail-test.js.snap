// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerDetail renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "marginTop": 25,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://images.punkapi.com/v2/keg.png",
          }
        }
        style={
          Object {
            "borderRadius": 10,
            "height": 250,
            "marginHorizontal": 10,
            "width": 250,
          }
        }
      />
      <View
        style={
          Object {
            "justifyContent": "space-around",
            "paddingTop": 25,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#666666",
              "fontSize": 19,
              "fontWeight": "bold",
              "padding": 10,
            }
          }
        >
           Name: 
          Buzz
        </Text>
        <Text
          style={
            Object {
              "color": "#666666",
              "fontSize": 19,
              "fontWeight": "bold",
              "padding": 10,
            }
          }
        >
           Description: 
          A light, crisp and bitter IPA brewed with English and American hops. A small batch brewed only once.
        </Text>
        <Text
          style={
            Object {
              "color": "#666666",
              "fontSize": 19,
              "fontWeight": "bold",
              "padding": 10,
            }
          }
        >
           Alcohol: 
          4.5
          %
        </Text>
        <Text
          style={
            Object {
              "color": "#666666",
              "fontSize": 19,
              "fontWeight": "bold",
              "padding": 10,
            }
          }
        >
           Food Pairings: 
          Spicy chicken tikka masala
          Grilled chicken quesadilla
          Caramel toffee cake
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
